docs/GLOSSARY.md

```diff
@@ -1,122 +1,180 @@
+
+An Action is not Dangerous if it is the only available Action which would allow them to progress, as in this scenario the Generator was unable to obtain other Checks even before they performed the Action. See also [Point of No Return](#point-of-no-return).
+
+## Dangerous Logic
+[Logic](#logic) that allows Actions to be performed which may place the game in an unbeatable state. Different from a [Point of No Return](#point-of-no-return) in that this is a primarily player-oriented issue; the game can still be beatable, but performing Actions in the wrong order can cause a player to experience this issue even though the [Seed](#seed) is valid.
+
+Examples:
+- Collecting the Gravity Suit in Metroid Prime, which makes Underwater Slope Jumps impossible
 
 ## Database
 
-TODO
+The sum total of all data that represents a game within Randovania. Contains a list of [Worlds](#world), a [Resource Database](#resource-database), a [Victory Condition](#victory-condition), and any other information necessary to represent the game.
+
+## Dock
+
+Represents some link between two [Areas](#area) that are directly adjacent. Docks usually represent things like doors, bridges, or other direct connections between areas, unlike [Teleporters](#teleporter), which represent means of travelling between two distant or disjointed Areas. Usually used to link two Areas within the same [World](#world), although it is possible for Docks to connect Areas between different Worlds.
 
 ## Event
 
-TODO
+A [Resource](#resource) representing something a player may do to access more of the game. Events can only be completed once, and cannot be undone.
+
+Examples:
+- Activating a switch that unlocks a door elsewhere
+- Defeating a boss which causes the states of other rooms to change
+
+## Experimental
+
+Any system or feature of Randovania which is not ready for public use.
 
 ## Game Patches
 
-TODO
+Has two meainings:
+
+- Colloquially, any modification made to the game. Usually refers to specific changes made to the game, such as cutscene alterations, and not to the randomization process itself.
+
+- In code, refers to the GamePatches class, which holds all per-layout game modifications for a single player. Crucially, this does not include cosmetic patches.
 
 ## Generator
 
-The part of Randovania that takes the [database](#database), the [preset](#preset) settings, a [random number generator](#rng) and decides which [pickups](#pickup) should be placed where, as well as everything else that is randomly decided in the [layout](#layout).
+The part of Randovania that takes the [Database](#database), the [Preset](#preset) settings, a [Random Number Generator](#rng) and decides how [Pickups](#pickup) should be shuffled in the [Layout](#layout). This is also where [Game Patches](#game-patches) and other game modifications are applied.
```

Comment:
Again, I would avoid mentioning explicitly it takes a database and preset because it's... incorrect! Since it's multiworld, there's multiple of these.

The last part is also wrong. They're not applied, it's where they are created in fact.

Reply:
The first part was the language of the document before I started modifying it, so I assumed it was correct.

Reply:
Rewrote this entirely.